python/cugraph/cugraph/experimental/datasets/dataset.py

```diff
@@ -0,0 +1,210 @@
+
+        filename = self.metadata['name'] + self.metadata['file_type']
+        if self._dl_path.path.is_dir():
+            df = cudf.read_csv(url)
+            df.to_csv(self._dl_path.path / filename, index=False)
+
+        else:
+            raise RuntimeError(f"The directory {self._dl_path.path.absolute()}"
+                               "does not exist")
+
+    def get_edgelist(self, fetch=False):
+        """
+        Return an Edgelist
+
+        Parameters
+        ----------
+        fetch : Boolean (default=False)
+            Automatically fetch for the dataset from the 'url' location within
+            the YAML file.
+        """
+
+        if self._edgelist is None:
+            full_path = self._dl_path.path / (self.metadata['name'] +
+                                              self.metadata['file_type'])
+
+            if not full_path.is_file():
+                if fetch:
+                    self.__download_csv(self.metadata['url'])
+                else:
+                    raise RuntimeError(f"The datafile {full_path} does not"
+                                       " exist. Try get_edgelist(fetch=True)"
+                                       " to download the datafile")
+
+            self._edgelist = cudf.read_csv(full_path,
+                                           delimiter=self.metadata['delim'],
+                                           names=self.metadata['col_names'],
+                                           dtype=self.metadata['col_types'])
+            self._path = full_path
+
+        return self._edgelist
+
+    def get_graph(self, fetch=False):
+        """
+        Return a Graph object.
+
+        Parameters
+        ----------
+        fetch : Boolean (default=False)
+            Automatically fetch for the dataset from the 'url' location within
+            the YAML file.
+        """
+        if self._edgelist is None:
+            self.get_edgelist(fetch)
+
+        self._graph = cugraph.Graph(directed=self.metadata['is_directed'])
+        self._graph.from_cudf_edgelist(self._edgelist, source='src',
+                                       destination='dst')
+
+        return self._graph
+
+    def get_path(self):
+        """
+        Returns the location of the stored dataset file
+        """
+        if self._path is None:
+            raise RuntimeError("Path to datafile has not been set." +
+                               " Call get_edgelist or get_graph first")
+
+        return self._path.absolute()
+
+
+def load_all(force=False):
+    """
+    Looks in `metadata` directory and fetches all datafiles from the the URLs
+    provided in each YAML file.
+
+    Parameters
+    force : Boolean (default=False)
+        Overwrite any existing copies of datafiles.
+    """
+    default_download_dir.path.mkdir(parents=True, exist_ok=True)
+
+    meta_path = Path(__file__).parent.absolute() / "metadata"
+    for file in os.listdir(meta_path):
```

Comment:
Here's another place you could use the `Path` methods:
```suggestion
    for file in meta_path.iterdir():
```
and `Path.iterdir()` returns `Path` objects.